### ISSUE TITLE
hashi_vault lookup plugin fails with:  KeyError: 'value'

### ISSUE THREAD

**Issue by techraf** [ansible/ansible#15495]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0 (devel c600ab81ee) last updated 2016/04/20 19:31:25 (GMT +900)
  lib/ansible/modules/core: (detached HEAD a38bdfc720) last updated 2016/04/02 18:12:29 (GMT +900)
  lib/ansible/modules/extras: (detached HEAD 7c3999a92a) last updated 2016/04/02 18:12:48 (GMT +900)
  config file = /Users/techraf/devops/playgrounds/vault-playground/ADHOC/ansible-integration/demo_02_pki_NG/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

Vault v0.5.2
##### OS / ENVIRONMENT

OS X 10.11.4 (15E65)
##### SUMMARY

The lookup plugin for `hashi_vault` fails to retrieve CA certificate from path /pki/cert/ca.

```
- debug: msg="{{ lookup('hashi_vault', 'secret=pki/cert/ca token={{ vault_token }} url=http://127.0.0.1:8200') }}"
```

At the same time the certificate is retrieved successfully with:
- command-line `vault read /pki/cert/ca`
- curl `curl -s -X GET http://127.0.0.1:8200/v1/pki/cert/ca`
- HVAC library `print(client.read('pki/cert/ca'))`

Also, lookup plugin for `hashi_vault` works OK for other secrets:

```
- debug: msg="{{ lookup('hashi_vault', 'secret=secret/hello token={{ vault_token }} url=http://127.0.0.1:8200') }}"
```
##### STEPS TO REPRODUCE

```
vault server -dev
export VAULT_ADDR='http://127.0.0.1:8200'
vault mount pki
vault mount-tune -max-lease-ttl=87600h pki
vault write pki/root/generate/internal common_name=myvault.com ttl=87600h
vault write pki/config/urls issuing_certificates="http://127.0.0.1:8200/v1/pki/ca" crl_distribution_points="http://127.0.0.1:8200/v1/pki/crl"
```

Then run the Ansible task:

```
- debug: msg="{{ lookup('hashi_vault', 'secret=pki/cert/ca token={{ vault_token }} url=http://127.0.0.1:8200') }}"
```
##### EXPECTED RESULTS

```
ok: [localhost] => {
    "msg": {
        "certificate": "-----BEGIN CERTIFICATE-----\nMIIDv....cut....\nrA==\n-----END CERTIFICATE-----\n"
    }
}
```
##### ACTUAL RESULTS

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/local/src/github.com/ansible/lib/ansible/executor/task_executor.py", line 124, in run
    res = self._execute()
  File "/usr/local/src/github.com/ansible/lib/ansible/executor/task_executor.py", line 390, in _execute
    self._task.post_validate(templar=templar)
  File "/usr/local/src/github.com/ansible/lib/ansible/playbook/task.py", line 246, in post_validate
    super(Task, self).post_validate(templar)
  File "/usr/local/src/github.com/ansible/lib/ansible/playbook/base.py", line 311, in post_validate
    value = templar.template(getattr(self, name))
  File "/usr/local/src/github.com/ansible/lib/ansible/template/__init__.py", line 358, in template
    d[k] = self.template(variable[k], preserve_trailing_newlines=preserve_trailing_newlines, fail_on_undefined=fail_on_undefined, overrides=overrides)
  File "/usr/local/src/github.com/ansible/lib/ansible/template/__init__.py", line 330, in template
    result = self._do_template(variable, preserve_trailing_newlines=preserve_trailing_newlines, escape_backslashes=escape_backslashes, fail_on_undefined=fail_on_undefined, overrides=overrides)
  File "/usr/local/src/github.com/ansible/lib/ansible/template/__init__.py", line 494, in _do_template
    res = j2_concat(rf)
  File "<template>", line 8, in root
  File "/usr/local/lib/python2.7/site-packages/jinja2/runtime.py", line 196, in call
    return __obj(*args, **kwargs)
  File "/usr/local/src/github.com/ansible/lib/ansible/template/__init__.py", line 415, in _lookup
    ran = instance.run(loop_terms, variables=self._available_variables, **kwargs)
  File "/usr/local/src/github.com/ansible/lib/ansible/plugins/lookup/hashi_vault.py", line 85, in run
    value = vault_conn.get()
  File "/usr/local/src/github.com/ansible/lib/ansible/plugins/lookup/hashi_vault.py", line 66, in get
    return data['data']['value']
KeyError: 'value'

fatal: [localhost]: FAILED! => {"failed": true, "msg": "Unexpected failure during module execution.", "stdout": ""}
```

**Comment by bcoca** [comment]
are you able to use it to get any other type of data successfully?

**Comment by techraf** [comment]
Either I don't understand the question or what I wrote in the issue description above already provides an answer:

> Also, lookup plugin for `hashi_vault` works OK for other secrets:
> 
> `debug: msg="{{ lookup('hashi_vault', 'secret=secret/hello token={{ vault_token }} url=http://127.0.0.1:8200') }}"`

If not, could you please give an example of what you asked for?

**Comment by Jorge-Rodriguez** [comment]
@techraf can you paste the output from the successful call using Vault's CLI? It seems to me that this particular secret does not have its content stored under the "value" key, remember that the secret values under a given path are pretty much free form.

**Comment by techraf** [comment]
Here are the results:

```
# vault read /pki/cert/ca
Key             Value
certificate     -----BEGIN CERTIFICATE-----
MIIDNTCCAh2gAwIBAgIUa484bUUWVEjmc4pULOk1pAjeJo8wDQYJKoZIhvcNAQEL
BQAwFjEUMBIGA1UEAxMLbXl2YXVsdC5jb20wHhcNMTYwNTE0MDY0NzM1WhcNMjYw
NTEyMDY0ODA1WjAWMRQwEgYDVQQDEwtteXZhdWx0LmNvbTCCASIwDQYJKoZIhvcN
AQEBBQADggEPADCCAQoCggEBAPg7yzC/1MHRii2ghYUS8D+U3m/5+hNH3aa9Mvl9
rToTvNYuv3di0DrJpqnq2EmCAJrEunNCbWYBGPYE5TQ6J1k6WDShhrnM6jtg2V2W
ql6zQL5kIAawWx4Ff5DoBSViJMTrHFJhe2g0xyr+RyuHU+Izk8BjZ7gqiv8rvZdc
jJoq96dQ+yenfU1e4ta4St35AgW2M0yTkkIy3k6JvnS25Q1u0L5jcOB0WWFOh8BF
ZMuBuM8nTe0eBBgy8kzPy8Wy5fJVs7Qv8ZBrcWWITEp8j6bGdXSFtWzNVE2zpR4L
9H+4FW2+ykJ1NPskvlXgreS40wtFOpG1dL2viTw3QWKMVQECAwEAAaN7MHkwDgYD
VR0PAQH/BAQDAgEGMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFC98njbAUxOf
Avt4J9nVCcMikRPTMB8GA1UdIwQYMBaAFC98njbAUxOfAvt4J9nVCcMikRPTMBYG
A1UdEQQPMA2CC215dmF1bHQuY29tMA0GCSqGSIb3DQEBCwUAA4IBAQBONhvPxBJf
A4C+HgX9CCr3IxBqwUjN/2m2yfXDyggZZ1Ga/N0JkxXRWDr7tTz24Bb4QNkCZ85l
Qj6bdrispuaNr0TVCUIACkY3jNGeokADwtcGCqPYruL11yCOUGlxORSLLp0e7VfX
hpzhx9qpIluA2ZnbQui/Tooc6br74Fv1utwjybFBCZASNPpzr0l+wzrcUruUYfOe
W12Ivk97Uej+ADMsxlvifFj5RjwVBL2aX4es/km0PAO2S5pzAzSgEw/5oZvmq2B0
hJZT6gLLNDaaUJsUxSwp9UusL+vbNTb9N5/JqWNF4N4+UwzhY2lx3UmTqo7jCmyo
5SOzKh2IsVGE
-----END CERTIFICATE-----
revocation_time 0
```

**Comment by Jorge-Rodriguez** [comment]
@techraf the certificate's value is under the `certificate` key, not under the `value` key, that is why the module is failing.

You could submit a feature request to be able to specify the key to retrieve, I have an improvement PR on this plugin, I'd be sure to add it.

**Comment by Jorge-Rodriguez** [comment]
Fixed on #13690

**Comment by bcoca** [comment]
closing via #13690
